custom_components/renault/sensor.py

```diff
@@ -57,172 +59,224 @@ async def get_entities(hass, proxy: PyzeProxy):
-            return data["chargingRemainingTime"]
-        LOGGER.debug("chargingRemainingTime not available in coordinator data %s", data)
-        return None
-
-
-class RenaultChargingPowerSensor(RenaultBatteryDataEntity):
-    """Charging Power sensor."""
-
-    @property
-    def state(self):
-        """Return the state of this entity."""
-        data = self.coordinator.data
-        if "chargingInstantaneousPower" in data:
-            if self.proxy.model_code in MODEL_USES_KWH:
-                return data["chargingInstantaneousPower"]
-            else:
-                return data["chargingInstantaneousPower"] / 1000
-        LOGGER.debug(
-            "chargingInstantaneousPower not available in coordinator data %s", data
-        )
-
-    @property
-    def unit_of_measurement(self):
-        """Return the unit of measurement of this entity."""
-        return POWER_KILO_WATT
-
-
-class RenaultOutsideTemperatureSensor(RenaultHVACDataEntity):
-    """HVAC Outside Temperature sensor."""
-
-    @property
-    def state(self):
-        """Return the state of this entity."""
-        data = self.coordinator.data
-        if "externalTemperature" in data:
-            return data["externalTemperature"]
-        LOGGER.debug("externalTemperature not available in coordinator data %s", data)
-
-    @property
-    def unit_of_measurement(self) -> str:
-        """Return the unit of measurement of this entity."""
-        return TEMP_CELSIUS
-
-
-class RenaultPlugStateSensor(RenaultBatteryDataEntity):
-    """Plug State sensor."""
-
-    @property
-    def state(self):
-        """Return the state of this entity."""
-        data = self.coordinator.data
-        if "plugStatus" in data:
-            try:
-                plug_state = PlugState(data["plugStatus"])
-            except ValueError:
-                plug_state = PlugState.NOT_AVAILABLE
-            return plug_state.name
-        LOGGER.debug("plugStatus not available in coordinator data %s", data)
-
-    @property
-    def icon(self):
-        """Icon handling."""
-        if self.state == PlugState.PLUGGED.name:
-            return "mdi:power-plug"
-        return "mdi:power-plug-off"
-
-
-class RenaultChargeStateSensor(RenaultBatteryDataEntity):
-    """Charge State sensor."""
-
-    @property
-    def state(self):
-        """Return the state of this entity."""
-        data = self.coordinator.data
-        if "chargingStatus" in data:
-            try:
-                charge_state = ChargeState(data["chargingStatus"])
-            except ValueError:
-                charge_state = ChargeState.NOT_AVAILABLE
-            return charge_state.name
-        LOGGER.debug("chargingStatus not available in coordinator data %s", data)
-
-    @property
-    def icon(self):
-        """Icon handling."""
-        if self.state == ChargeState.CHARGE_IN_PROGRESS.name:
-            return "mdi:flash"
-        return "mdi:flash-off"
-
+if self.model_code in MODEL_USES_KWH:
```

Comment:
What's the point to of this condition? That's unusual to avoid class declaration.